draft-ietf-wimse-s2s-protocol.md

```diff
@@ -401,11 +401,17 @@ To validate the WPT in the request, the recipient MUST ensure the following:
 * Optionally, check that the value of the `jti` claim has not been used before in the time window in which the
  respective WPT would be considered valid.
 * If presented in conjunction with an OAuth access token, the value of the `ath` claim matches the hash of that token's value.
-* If presented in conjunction with a Txn-Token, the value of the `tth` claim matches the hash of that token's value.
-* If presented in conjunction with a token conveying end-user identity or authorization context, the value of
- the `oth` claim matches the hash of that token's value.
+* If `Workload-Proof-Token` contains `signed_headers` claim:
+   * The request contains a header for each claim in the `signed_headers`.
+   * Each corresponding value in the claim matches the hash of the value in the header.
+   * Headers not present in the `Workload-Proof-Token` are accepted without further proof.
+   * If a header is present more than once in the request, each value has a corresponding hash in the `Workload-Proof-Token`. Proof of just a single value in a multi-value header are not allowed.
```

Comment:
I believe (but don't consider myself an HTTP expert) that if a header is repeated multiple times, this is equivalent to concatenating the values together. Oh, here's the text from RFC 9421 again (@bc-pi will love it, I'm sure):

> Unless overridden by additional parameters and rules, HTTP field values MUST be combined into a single value as defined in [Section 5.2](https://rfc-editor.org/rfc/rfc9110#section-5.2) of [[HTTP](https://www.rfc-editor.org/rfc/rfc9421.html#RFC9110)] to create the component value. Specifically, HTTP fields sent as multiple fields MUST be combined by concatenating the values using a single comma and a single space as a separator ("," + " "). Note that intermediaries are allowed to combine values of HTTP fields with any amount of whitespace between the commas, and if this behavior is not accounted for by the verifier, the signature can fail, since the signer and verifier will see a different component value in their respective signature bases. For robustness, it is RECOMMENDED that signed messages include only a single instance of any field covered under the signature, particularly with the value for any list-based fields serialized using the algorithm below. This approach increases the chances of the field value remaining untouched through intermediaries. Where that approach is not possible and multiple instances of a field need to be sent separately, it is RECOMMENDED that signers and verifiers process any list-based fields taking all individual field values and combining them based on the strict algorithm below, to counter possible intermediary behavior.

Reply:
This solution, which shouldn't attempt to be a generic signature thing, should probably constrain itself to single value things.